File: HOT100/14.贪心算法/45.跳跃游戏-ii.py
#
# @lc app=leetcode.cn id=45 lang=python3
#
# [45] 跳跃游戏 II
#
from mytools import *
# @lc code=start
class Solution:
    def jump(self, nums: List[int]) -> int:
        cur_co, nxt_co = 0, 0
        if len(nums) == 1: return 0
        res = 0
        for i in range(len(nums)):
            nxt_co = max(i + nums[i], nxt_co)
            if i == cur_co:
                res += 1
                cur_co = nxt_co
                if cur_co >= len(nums) - 1: break
        return res
# @lc code=end

File: HOT100/14.贪心算法/45.跳跃游戏-ii.cpp
/*
 * @lc app=leetcode.cn id=45 lang=cpp
 *
 * [45] 跳跃游戏 II
 */
#include "tools.h"
// @lc code=start
class Solution {
public:
    int jump(vector<int>& nums) {
        int cur_co = 0;
        int nxt_co = 0;
        int res = 0;
        if (nums.size() == 1) return 0;
        for (int i = 0; i < nums.size(); i ++ )
        {
            nxt_co = max(nxt_co, i + nums[i]);
            if (i == cur_co)
            {
                cur_co = nxt_co;
                res ++;
                if (cur_co >= nums.size() - 1) return res;
            }
        }
        return res;
    }
};
// @lc code=end

